### PR TITLE
proxy-11: add endpoint to process user data and return picture of rover

### PR DESCRIPTION
**Pull request by plutsenka**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,6 +28,7 @@
     "dotenv": "^16.4.5",
     "express": "^4.18.3",
     "http-proxy-middleware": "^2.0.6",
-    "moment": "^2.30.1"
+    "moment": "^2.30.1",
```
Comment (plutsenka) [inline package.json]: moment was removed in previous PR, so I left it here to prevent conflicts

File: src/routes/userData.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+const express = require('express');
+const router = express.Router();
+const userService = require('../services/userData');
+const roverPhotoService = require('../services/roverPhotos');
+const UserRequestDto = require('../dto/userRequestDto');
+
+router.post('/', async function (req, res, next) {
+  const userData = new UserRequestDto(req.body);
```
Comment (mdziarkach) [inline src/routes/userData.js:8]: point for improvements: ideally It is better to validate data in the middleware before the controller. If the data is not valid, throw an exception

File: src/services/roverPhotos.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+const axios = require('axios');
+const { format, subDays } = require('date-fns');
+require('dotenv').config();
```
Comment (mdziarkach) [inline src/services/roverPhotos.js]: try using this line only in the root file

File: src/services/roverPhotos.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+const { API_BASE_URL, MARS_PHOTO_ENDPOINT, MANIFESTS_ENDPOINT } = process.env;
+const { API_KEY } = process.env;
```
Comment (mdziarkach) [inline src/services/roverPhotos.js]: could you create config file with all env variables?
in the future you will be able to validate variables

File: src/app.js [hunk 5]
```diff
@@ -1,12 +1,16 @@
 const express = require('express');
 const meteorsDataRouter = require('./routes/meteorsData');
+const userDataRouter = require('./routes/userData');
 require('dotenv').config();
 
 const app = express();
 
 const { PORT } = process.env;
```
Comment (mdziarkach) [inline src/app.js]: const port = process.env.PORT || 4000

File: src/routes/userData.js (hunk 2 above)
Comment (plutsenka) [reply src/routes/userData.js:8]: Will implement validation as a part of future work (I saw there is a task for this)

**Review by mdziarkach — APPROVED** [review]
Looks better

**Pull request merged**